firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -485,14 +480,16 @@ private HttpURLConnection openHttpURLConnection(URL url, String apiKey)
     httpURLConnection.addRequestProperty(CONTENT_ENCODING_HEADER_KEY, GZIP_CONTENT_ENCODING);
     httpURLConnection.addRequestProperty(CACHE_CONTROL_HEADER_KEY, CACHE_CONTROL_DIRECTIVE);
     httpURLConnection.addRequestProperty(X_ANDROID_PACKAGE_HEADER_KEY, context.getPackageName());
-    if ((heartbeatInfo.get() != null) && (userAgentPublisher.get() != null)) {
-      HeartBeat heartbeat =
-          heartbeatInfo.get().getHeartBeatCode(FIREBASE_INSTALLATIONS_ID_HEARTBEAT_TAG);
-      if (heartbeat != HeartBeat.NONE) {
-        httpURLConnection.addRequestProperty(
-            USER_AGENT_HEADER, userAgentPublisher.get().getUserAgent());
```

Comment:
No more userAgent in the heartbeat?

Reply:
this is included in the HEARTBEAT_HEADER itself now. So we don't require two keys. Basically the HEARTBEAT_HEADER is the base64 encoding of a dictionary with userAgent as the key and the set of dates when firebase was used with that useragent string as the value. More information on this can be found at go/firebaseandroid-heartbeat